### ISSUE TITLE
np array slicing issue

### ISSUE THREAD

**Issue by andyvandy** [cunnane/xloil#86]
I've spotted an issue with slicing on win32bit, python 3.11.4, xloil==0.18.4, numpy==1.26.3

I've narrowed it down to being related to `xlo.Array(dims=1)`  and have put together a minimal repro below. When I provide a step, the slice doesn't step correctly and grabs the first `n/step` elements and if the slice is negative, excel crashes. 

I'll circle back later to investigate further. Please don't feel the need to look into this until I ( or another user! ) do a bit more digging though. 

Again, thanks for making the project available!!

Sample excel calls

```
=e_slice({1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20},,,2)
-> output= {1;3;5;7;9;11;13;15;17;19}

=np_slice_repro({1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20},,,2)
-> {1;2;3;4;5;6;7;8;9;10}

=np_slice({1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20},,,2)
-> output= {1;3;5;7;9;11;13;15;17;19}
```
sample xlo funcs
```python
import xloil as xlo
def sanitize_inputs(*args):
    """Could sort out a converter in the future"""
    return [arg if arg is None else int(arg) for arg in args]

def slice_fn(arr, *args):
    start,end,step = sanitize_inputs(*args)
    return arr[start:end:step ]

@xlo.func(local=False)
def e_slice(arr, start=None, end=None, step=None): #works
    return( slice_fn(arr,start,end,step))

@xlo.func(local=False)
def np_slice(arr:xlo.Array(), start=None, end=None, step=None):#works
    return( slice_fn(arr,start,end,step))

@xlo.func(local=False)
def np_slice_repro(arr:xlo.Array(dims=1), start=None, end=None, step=None): #breaks
    return( slice_fn(arr,start,end,step))

```

**Comment by andyvandy** [comment]
If I change the repro to the following, it resolves the issue:
```python
def np_slice_repro(arr:xlo.Array(int, dims=1), start=None, end=None, step=None)-> xlo.Array(int, dims=1): 
```

This is linked to [numpy array views ](https://numpy.org/doc/stable/user/basics.copies.html) which are implementing the slice by simple modifying the stride rather than touching the underlying data.

I believe the root issue is a difference in the implementation of the `operator()` function for `XlFromArray1d` based on content type 

in ` XlFromArray1d<NPY_OBJECT>` 
https://github.com/cunnane/xloil/blob/935d7a5d850b3897dee9ddfabff372c5f533509f/libs/xlOil_Python/TypeConversion/Numpy.cpp#L739-L747
vs in `template <int TNpType> XlFromArray1d `
https://github.com/cunnane/xloil/blob/935d7a5d850b3897dee9ddfabff372c5f533509f/libs/xlOil_Python/TypeConversion/Numpy.cpp#L657-L665


I think aligning the implementations to both use `PyArray_STRIDE` should resolve this.

**Comment by cunnane** [comment]
Thanks, I do like an issue report containing the line number of the bug!  This is fixed in 0.18.5 which I just released

**Comment by andyvandy** [comment]
Just tested this out and checked the commit. lgtm. 

Thanks for including this so quickly!
